grakel_replace/optimize.py

```diff
@@ -81,21 +116,34 @@ def optimize_acqf_graph(
         raise ValueError("train_graphs cannot be None.")
 
     sampled_graphs = sample_graphs(train_graphs, num_samples=num_graph_samples)
+    gp = acq_function.model
+    covar_module = gp.covar_module
 
     best_candidates, best_scores = [], []
 
+    TODO_GRAPH_COLUMN_INDEX = bounds.shape[1] - 1
+
     for _graph in sampled_graphs:
-        for fixed_features in fixed_features_list or [{}]:
-            candidates, scores = optimize_acqf_mixed(
-                acq_function=acq_function,
-                bounds=bounds,
-                fixed_features_list=[fixed_features],
-                num_restarts=num_restarts,
-                raw_samples=raw_samples,
-                q=q,
-            )
-            best_candidates.append(candidates)
-            best_scores.append(scores)
+        # This is new, we essentially iterate through all the kernels and
+        # include the sampled graph.
+        with set_graph_lookup(covar_module, [_graph], append=True):
+            for fixed_features in fixed_features_list or [{}]:
+                # We then consider this graph as a fixed feature, i.e. in the X's
+                # generated during acquisition, the graph column will just be full
+                # of `-1` indicating to select the very last graph in the lookup
+                # they used.
+                _fixed_features = {**fixed_features, TODO_GRAPH_COLUMN_INDEX: -1.0}
```

Comment:
Here's where we tell botorch to set the graph column for it's acqusition samples to be constant at `-1`, i.e. take the last `_graph`.